### PR TITLE
Minor housekeeping

### PR DESCRIPTION
**Pull request by andrew-boyarshin**
* Fix never showing error dialog (result not assigned to bool variable)
* Add `nullptr` checks to `malloc` & `calloc` calls
* Fix `std::string` -> `char*` -> `std::string` conversions
* Add `std::move` to reduce data copying
* Catch C++ exception by reference

### CODE REVIEW

File: src/program/OutfitProject.cpp [hunk 1]
```diff
@@ -1135,19 +1135,19 @@ void OutfitProject::SetTextures(NiShape* shape, const std::vector<std::string>&
 
 			if (!mat.Failed()) {
 				if (mat.signature == MaterialFile::BGSM) {
-					texFiles[0] = mat.diffuseTexture.c_str();
```
Comment (ousnius) [inline src/program/OutfitProject.cpp]: The c_str() is needed to skip duplicate NULL terminators that were added to the strings from IO. Without the c_str(), the strings will end up at a size of 1 with \0 as the explicit content, which will make my !texFiles[i].empty() check a few lines below fail.

Until I find a way to fix the \0 terminators in IO after reading the material files, this change has to be removed (same in PreviewWindow.cpp). I might need to resize the string to c_str() after reading it from the file.

File: src/program/PreviewWindow.cpp [hunk 2]
```diff
@@ -246,19 +246,19 @@ void PreviewWindow::AddNifShapeTextures(NifFile* fromNif, const std::string& sha
 
 		if (!mat.Failed()) {
 			if (mat.signature == MaterialFile::BGSM) {
-				texFiles[0] = mat.diffuseTexture.c_str();
```
Comment (ousnius) [inline src/program/PreviewWindow.cpp]: Same here.

**Review by ousnius — CHANGES_REQUESTED** [review]
See review comments. Otherwise seems fine, thanks.

**Comment by andrew-boyarshin** [comment]
@ousnius reverted those changes to your request.

**Review by ousnius — APPROVED** [review]

**Pull request merged**